graftm/hmmer.py

```diff
@@ -507,56 +500,57 @@ def alignment_correcter(self, alignment_file_list, output_file_name):
+        orfm: obj
+            graftm.OrfM object with parameters already set
+        orf_out_path
+            Path to output fasta file, containing amino acid ORFs
         '''
-        'Extract only the orfs that hit the hmm, return sequence file with within.'        
-        # Build the command
-        output_table_list = []
-        if len(self.search_hmm) > 1:
-            for hmm in self.search_hmm:
-                out = os.path.join(os.path.split(hmmsearch_out_path)[0], os.path.basename(hmm).split('.')[0] +'_'+ os.path.split(hmmsearch_out_path)[1])
-                output_table_list.append(out)
-        elif len(self.search_hmm) == 1:
-            output_table_list.append(hmmsearch_out_path)
+        raw_orf_path    = tempfile.NamedTemporaryFile(prefix='orf_hmmsearch').name
+        orf_titles_path = tempfile.NamedTemporaryFile(prefix='orf_titles').name
+        
+        # Build the output_files
+        if len(self.search_hmm) >= 1:
+            output_table_list=[tempfile.NamedTemporaryFile(prefix='orf_hmmsearch').name for x in self.search_hmm]
         else:
             raise Exception("Programming error: expected 1 or more HMMs")
-        
-        # Call orfs on the sequences
+
+        # Build ORF calling command orfs on the sequences
         orfm_cmd = orfm.command_line()
         cmd = '%s %s > %s' % (orfm_cmd, input_path, raw_orf_path)
         
         logging.debug("Running command: %s" % cmd)
         subprocess.check_call(cmd, shell=True)
-
+        
         cmd = 'cat %s' % raw_orf_path
         searcher = HmmSearcher(1)
         searcher.hmmsearch(cmd, self.search_hmm, output_table_list)
-        
         with open(orf_titles_path, 'w') as output:
-            seen = []
-            for table in output_table_list:
-                for title in [x.split(' ')[0] for x in open(table).readlines() if not x.startswith('#')]:
-                    if title not in seen:
-                        output.write(str(title) + '\n')
-                        seen.append(title)
-                    else:
-                        pass       
-        
+            reads= set(
+                       sum([HMMreader(x).names() for x in output_table_list], 
+                           [])
+                       )
+            [output.write(x + '\n') for x in reads]
+            
         # Extract the reads using the titles.
         cmd = 'fxtract -H -X -f %s %s > %s' % (orf_titles_path, raw_orf_path, orf_out_path)
-        
         logging.debug("Running command: %s" % cmd)
         subprocess.check_call(cmd, shell=True)
-        
-        # Return name of output file
-        return orf_out_path
-
-    def p_search(self, files, args, run_stats, base, unpack, raw_reads):
-        '''Protein search pipeline - The searching step for the protein 
+    
+    @timeit
+    def p_search(self, files, args, base, unpack, raw_reads, search_method, gpkg):
+        '''
+        Protein search pipeline - The searching step for the protein 
```

Comment:
"Protein" is confusing here, I never remember whether you mean searching protein sequences or searching with a protein HMM/diamonddb